src/chain_replica.mjs

```diff
@@ -1,17 +1,15 @@
-/* global Buffer */
-
-import fs from 'fs';  // ISSUE: AMBIENT
+// Reference to TypeScript definitions for IntelliSense in VSCode
+/// <reference path="../rnode-grpc-gen/js/rnode-grpc-js.d.ts" />
 
 import postgres from 'postgres';  // ISSUE: AMBIENT
 import grpcLib from '@grpc/grpc-js'; //@@ AMBIENT
 
-import rnode_grpc_js from '@tgrospic/rnode-grpc-js';
-// requires --experimental-json-modules
-import protoSchema from '../rchain-proto/rnode-grpc-gen/js/pbjs_generated.json';
-import '../rchain-proto/rnode-grpc-gen/js/DeployServiceV1_pb.js'; // proto global
-
-const { signDeploy, rnodeDeploy, getAddrFromPrivateKey } = rnode_grpc_js;
+// RNode with environment parameters
+import { rnodeService } from './rnode-env.mjs';
 
+// Load .env file
+import { config } from 'dotenv';
+config();
```

Comment:
ambient authority grumble...

Reply:
I don't have any nice solution for this. dotenv uses fs and path directly and I didn't find some more SES compatible variant.

Do you have any suggestions? Is it possible to use SES shim to load it inside separate Compartment with provided fs and path as only global variables?

Reply:
> ... dotenv uses fs and path directly ...

yeah; that's a bug. reimplementing dotenv with ocap discipline is probably about 20 lines of code. But it can wait for a later PR.

Reply:
> Is it possible to use SES shim to load it inside separate Compartment with provided fs and path as only global variables?

Yes, that can work too, if you _really_ have to use 3rd party code and you can't change it.

Reply:
> yeah; that's a bug. reimplementing dotenv with ocap discipline is probably about 20 lines of code. But it can wait for a later PR.

:+1: I wonder should it be update of existing lib or a new package?

---

I tried to import dotenv in isolated Compartment.

```js
import 'ses';
lockdown();

const c = new Compartment({fs, path});

const r = c.evaluate(`
  import { config } from 'dotenv';
  config();
`);
```
But I see this error:
```
(SyntaxError#1)
SyntaxError#1: Cannot use import statement outside a module

  at Object.eval (eval at makeEvaluateFactory (file:///home/tomi/projects/rchain/rchat/node_modules/ses/dist/ses.umd.js:1400:10), <anonymous>:8:30)
```

Reply:
Issue to track updates #12.